### ISSUE TITLE
Early Desync (798)

### ISSUE THREAD

**Issue by GoogleFrog** [beyond-all-reason/spring#191]
Happens within a minute.

Tim_Wolfman http://zero-k.info/Battles/Detail/1287518

https://github.com/ZeroK-RTS/CrashReports/issues/52874
https://github.com/ZeroK-RTS/CrashReports/issues/52872
https://github.com/ZeroK-RTS/CrashReports/issues/52865
https://github.com/ZeroK-RTS/CrashReports/issues/52855
https://github.com/ZeroK-RTS/CrashReports/issues/52870

[20220123_204418_FrozenFortress_v2_105.1.1-798-gae7f0f8 BAR105.sdfz.txt](https://github.com/beyond-all-reason/spring/files/7921706/20220123_204418_FrozenFortress_v2_105.1.1-798-gae7f0f8.BAR105.sdfz.txt)

**Comment by GoogleFrog** [comment]
Within four minutes

http://zero-k.info/Battles/Detail/1287424

https://github.com/ZeroK-RTS/CrashReports/issues/52845
https://github.com/ZeroK-RTS/CrashReports/issues/52839
https://github.com/ZeroK-RTS/CrashReports/issues/52868

[20220123_185922_Small Supreme Battlefield V2_105.1.1-798-gae7f0f8 BAR105.sdfz.txt](https://github.com/beyond-all-reason/spring/files/7921725/20220123_185922_Small.Supreme.Battlefield.V2_105.1.1-798-gae7f0f8.BAR105.sdfz.txt)

**Comment by lhog** [comment]
With desyncs generally a few basic diagnostics are possible:
1. A desyncee runs the replays and sees if he gets the same (desync) checksum as in the game. If not, then it's likely one time off bit flip or something like that.
2. A number of us, devs, can run replay and see if they can the same checksum as the desyncee or different checksum than everybody else.
3. I can run the replay with asserts on in almost hopeless expectation something will trigger.

2. and 3. are the best outcomes cause we can go iteratively, 1. will likely require user participation and a few days of time.

**Comment by lhog** [comment]
I checked both desyncs and they didn't trigger any asserts, so practically only two options left.
* One of us hit the same desync checksum as desyncee from the demo, that'd be incredible luck based on the previous experience (practically never happened before)
* We can get hold of a desyncee, make them run the demo and see if they desync in the demo again. Such desynced person will need to spend considerable time with us swapping executables, we are going to build and uploading multimegabytes traces.

In the meantime @ashdnazg fixed compiler dependent assimp desync, that prevented debugging of Zero-K games on Visual Studio. I also found forgotten springcontent files leading to early desync with my local debug build. This update is already deployed on ZK server. So we will need a few more early desync cases with new engine + passionate person to get debug done with.

In the meantime I'd be cool to rethink what SYNCDEBUG used to do (now it's reportedly broken) and perhaps revive it.
